unyt/unit_object.py

```diff
@@ -480,18 +480,16 @@ def __pow__(self, p):
 
     def __eq__(self, u):
         """Test unit equality."""
-        if not isinstance(u, Unit):
-            return False
         return (
-            math.isclose(self.base_value, u.base_value)
+            isinstance(u, Unit)
             and self.dimensions == u.dimensions
+            and math.isclose(self.base_value, u.base_value)
+            and math.isclose(self.base_offset, u.base_offset)
         )
 
     def __ne__(self, u):
         """Test unit inequality."""
-        if not isinstance(u, Unit):
-            return True
-        if not math.isclose(self.base_value, u.base_value):
+        if not self == u:
             return True
         # use 'is' comparison dimensions to avoid expensive sympy operation
```

Comment:
clearly I'm not taking advantage of these optimisations yet. I'll circle back to improving code sharing VS optimization if the principle of the PR is deemed acceptable